spec/csv_row_model/integrations/dynamic_columns_spec.rb

```diff
@@ -0,0 +1,107 @@
+require 'spec_helper'
+
+describe 'Dynamic columns scenario' do
+
+
+  context 'import' do
+
+    class ImportDynamicColumnModel < DynamicColumnModel
+      include CsvRowModel::Import
+
+      def self.skills_request
+        ['Clean', 'Punctual'].map { |skill_name| SkillModel.new(skill_name, true) }
+      end
+
+      def self.skills_request_header
+        skills_request.map { |skill| skill.name }
+      end
+    end
+
+    let(:csv_source) do
+      [
+        [ 'First Name' , 'Last Name'  , 'Organize', 'Clean', 'Punctual', 'Strong', 'Crazy', 'Flexible' ],
+        [ 'Josie'      , 'Herman'     , 'Yes'     , 'Yes'  , 'No'      , 'No'    , 'Yes'  , 'No'       ],
+        [ 'Florian'    , 'Bartoletti' , 'Yes'     , 'Yes'  , 'No'      , 'Yes'   , 'Yes'  , 'No'       ],
+        [ 'Eriberto'   , 'Considine'  , 'Yes'     , 'Yes'  , 'Yes'     , 'No'    , 'Yes'  , 'No'       ],
+        [ 'Mathias'    , 'Durgan'     , 'Yes'     , 'Yes'  , 'No'      , 'No'    , 'Yes'  , 'No'       ],
+      ]
+    end
+
+    let(:csv_string) do
+      CSV.generate do |csv|
+        csv_source.each { |row| csv << row }
+      end
+    end
+
+    let(:file) do
+      file = Tempfile.new(['input_file','.csv'])
+      file.write(csv_string)
+      file.rewind
+      file
+    end
+
+    subject do
+      CsvRowModel::Import::File.new(file.path, ImportDynamicColumnModel, {})
+    end
+
+    it '' do
+      row_model = subject.next
+
+      expect(row_model.first_name).to eql('Josie')
+      expect(row_model.last_name).to eql('Herman')
+      expect(row_model.skills).to be_an(Array)
+      expect(row_model.skills.map(&:name)).to eql(['Clean', 'Punctual'])
+      expect(row_model.skills.map(&:to_s)).to eql(['Yes', 'Yes'])
+      expect(row_model.class.headers).to eql(['First Name', 'Last Name', 'Clean', 'Punctual'])
+    end
+  end
+
+  context 'export' do
+
+    class ExportDynamicColumnModel < DynamicColumnModel
+      include CsvRowModel::Export
+
+      # NOTE Should dynamic and speak with instance.
+      def self.skills_request_header
+        ['Organize', 'Clean', 'Punctual', 'Strong', 'Crazy', 'Flexible']
```

Comment:
here as well we need to talk with `source_model` to do something like `source_model.data.map(&:to_s)`